src/dist/manifest.rs

```diff
@@ -47,16 +47,34 @@ pub enum PackageTargets {
 
 #[derive(Clone, Debug, PartialEq)]
 pub struct TargetedPackage {
-    pub bins: Option<PackageBins>,
+    pub bins: Vec<(CompressionKind, HashedBinary)>,
```

Comment:
Rather than a tuple, what about a data carrying enum ?

```
GZip(HashedBinary),
XZ(HashedBinary),
ZStd(HashedBinary)
```

Or alternatively, a field in HashedBinary:

`pub kind CompressionKind,`
I'm not sure it would turn out better, but I think if we always want to know we shouldn't rely on storing them adjacently in a vector

Reply:
I would have stored this as a `HashMap<CompressionKind, HashedBinary>` but for the fact that (a) HashMap is not PartialEq, and (b) I wanted to retain a preference order in there.  As such, given you don't feel the need to block on this, I'm going to merge.

Reply:
As for an enum which carries data, that makes storing a preference order in a const a bit harder :D

Reply:
So there are two separate concerns here which I think you've tangled up.

One is the knowledge that there are serialised prefixes in the external table.
The other is the tangled threads through the callers of the kind, which is shown by the places this touched and the (kind, url, path) tuple later on.

We don't necessarily need one kind for both of these things, and the preference order is only relevant, if at all, for the former.